### PR TITLE
Document a contrived example for EmberArray#filter target option

### LINKED ISSUE #18522: EmberArray#filter - Document the optional target object parameter
Reference: https://github.com/emberjs/ember.js/pull/18239/files#r314775795

### PR DESCRIPTION
**Pull request by mehulkar**
Going through the Documentation label and picking off some easy ones: https://github.com/emberjs/ember.js/issues?q=is%3Aopen+is%3Aissue+label%3ADocumentation

### CODE REVIEW

File: packages/@ember/-internals/runtime/lib/mixins/array.js [hunk 1]
```diff
@@ -870,7 +870,27 @@ const ArrayMixin = Mixin.create(Enumerable, {
...
+
+      constructor(opts = {}) {
+        super(...arguments);
+
+        this.engineering = opts.engineering;
+        this.people = Ember.A([{ name: 'John', age: 14 }, { name: 'Joan', age: 45 }]);
+      }
+    }
+
+    const collection = new AdultsCollection({ engineering: true });
```
Comment (locks) [inline packages/@ember/-internals/runtime/lib/mixins/array.js]:
```suggestion
    let collection = new AdultsCollection({ engineering: true });
```

**Review by locks — CHANGES_REQUESTED** [review]
one small ask. ember resources generally use ["liberal let"](https://madhatted.com/2016/1/25/let-it-be) style.

**Comment by mehulkar** [comment]
np, updated :)

**Review by locks — APPROVED** [review]

**Comment by locks** [comment]
Thanks!

**Pull request merged**